pkg/monitor/cluster/nodes_test.go

```diff
@@ -0,0 +1,65 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitNodesMetrics(t *testing.T) {
+	cli := fake.NewSimpleClientset(&corev1.Node{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "node",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "aro-master-0",
+		},
+		Status: corev1.NodeStatus{
+
+			Conditions: []corev1.NodeCondition{
+				{
+					Type:   corev1.NodeMemoryPressure,
+					Status: corev1.ConditionTrue,
+				},
+			},
+		},
+	}, &corev1.Node{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "node",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "aro-master-1",
+		},
+		Status: corev1.NodeStatus{
+
+			Conditions: []corev1.NodeCondition{
+				{
+					Type:   corev1.NodeReady,
+					Status: corev1.ConditionFalse,
+				},
+			},
+		},
+	})
+
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	m := mock_metrics.NewMockInterface(controller)
+	mon := &Monitor{
+		dims: map[string]string{},
+		cli:  cli,
+		m:    m,
+	}
+	m.EXPECT().EmitGauge("cluster.nodes.total", int64(2), map[string]string{})
+	m.EXPECT().EmitGauge("cluster.nodes.condition", int64(1), map[string]string{"condition": "NotReadyNodes"})
+	m.EXPECT().EmitGauge("cluster.nodes.condition", int64(1), map[string]string{"condition": "MemoryPressureNodes"})
```

Comment:
Do we also want to check for e.g. DiskPressureNodes = 0 to ensure that metric's being emitted?

Reply:
@ehashman the requirement is to not emit metrics if the count of nodes for that condition is 0. I haven't seen a test construct to express "expect" not been called.

Reply:
@ehashman there is actually no need in "not expect". The framework works in such a way that if something was called but not was declared with `EXPECT()`, the test will fail. So this test guarantees that there were exactly 3 calls to that function.